src/diagnostic-channel-publishers/src/pg.pub.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import {channel, IModulePatcher, PatchFunction} from "diagnostic-channel";
+import {EventEmitter} from "events";
+
+// copy the pg.Result type: https://node-postgres.com/api/result
+export interface IPostgresResult {
+    rowCount: number;
+    command: string;
+}
+
+export interface IPostgresData {
+    query: {
+        text?: string;
+        plan?: string;
+        preparable?: {
+            text: string;
+            args: any[];
+        }
+    };
+    database: {
+        host: string;
+        port: string;
+    };
+    result?: IPostgresResult;
+    duration: number;
+    error?: Error;
+}
+
+type PostgresCallback = (err: Error, res: IPostgresResult) => any;
+
+function postgres6PatchFunction(originalPg, originalPgPath) {
+    const originalClientQuery = originalPg.Client.prototype.query;
+
+    // this keeps track of user-supplied callback => patched callbacks so we don't keep
+    // re-patching the same callback over and over for library functions
+    // IMPORTANT: use a WeakMap so that we don't keep references to temporary/otherwise unreferenced functions
+    const patchedCallbacks = new WeakMap<PostgresCallback, PostgresCallback>();
+
+    // wherever the callback is passed, find it, save it, and remove it from the call
+    // to the the original .query() function
+    originalPg.Client.prototype.query = function query(config, values, callback) {
+        const data: IPostgresData = {
+            query: {},
+            database: {
+                host: this.connectionParameters.host,
+                port: this.connectionParameters.port,
+            },
+            result: null,
+            error: null,
+            duration: 0,
+        };
+        const start = process.hrtime();
+        let queryResult;
+
+        function patchCallback(cb?: PostgresCallback, cachePatched?: boolean): PostgresCallback {
+            const existingPatch = patchedCallbacks.get(cb);
+            if (existingPatch) {
+                // edge case where a previously-seen user callback is now being used
+                // as the callback of a config object
+                if (cachePatched) {
+                    patchedCallbacks.set(existingPatch, existingPatch);
+                }
+
+                return existingPatch;
+            }
+
+            const trackingCallback = channel.bindToContext(function(err: Error, res: IPostgresResult): any {
+                const end = process.hrtime(start);
+                data.result = res && { rowCount: res.rowCount, command: res.command };
```

Comment:
res.command is always the command "type" (at least according to [pg v7](https://node-postgres.com/api/result#-code-result-command-string-code-), I can double-check v6 but I don't remember this changing much), so in this case it would be just the string "ALTER"